app/models/vm_or_template.rb

```diff
@@ -31,6 +31,8 @@ class VmOrTemplate < ApplicationRecord
 
   include AvailabilityMixin
 
+  supports_not :retirement, ""
```

Comment:
I've pushed a commit for that. but seems @h-kataria is manually incorporating my PR into hers.